### PR TITLE
Unpowered material for the medistation

### PR DESCRIPTION
**Pull request by illwieckz**
Unpowered material for the medistation, fixes:

- https://github.com/Unvanquished/Unvanquished/issues/2598

Also fix a minor discrepancy in armoury material.

### CODE REVIEW

File: scripts/medistat.shader [hunk 1]
```diff
@@ -39,6 +54,7 @@ models/buildables/medistat/display
 		blendfunc add
 	}
 	when idle2 models/buildables/medistat/displayH
+	when unpowered textures/shared_colors/black
```
Comment (ghost) [inline scripts/medistat.shader]:
```suggestion
when unpowered models/buildables/arm/screen
```

Well, I'd put this image into a generic shader too, ofc, but the black texture is really ugly.

**Review by ghost — CHANGES_REQUESTED** [review]
Just don't use a black texture for screen, it's ugly. Otherwise: tested and approved.

**Comment by ghost** [comment]
Regarding my comment:



Guess which one is using the black texture?

Also, I see the middle thingy is still "powered" but might be the model?

File: scripts/medistat.shader (hunk 1 above)
Comment (illwieckz) [reply scripts/medistat.shader]: Ok then.

**Comment by illwieckz** [comment]
I removed the black texture.

> Also, I see the middle thingy is still "powered" but might be the model?

Well, we only disable the glow map, we could make another diffuseMap for the dead medistation.

**Review by ghost — DISMISSED** [review]

**Comment by ghost** [comment]
I think we could use armoury's screen for that, but it's good enough like this:



As a side note, I wonder if we should not remove the glowmap of the medistat... but not the scope of this.

**Comment by ghost** [comment]
I added commit to use arm screen as unpowered display for medistat (see previous screenshots to see the difference). I can not do anything about the bright dot though, this requires modifying png and I don't know how to make something that would look good.

**Comment by mjstahlberg** [comment]
Comparison at this point in time:

Powered:
unvanquished_2023-04-08_084457_000

Unpowered:
unvanquished_2023-04-08_084507_000

This is much better than before and could be merged. <s>The remaining problem is that the diffuse map was not designed to be used without the glow map, using the same colors.</s> This is particularly annoying around the screen where you can see blue edges around the now-gray unpowered screen (zoom in). I will briefly check if I'm able to fix that.

Edit: Looking at the sources, I now see that the texture artist *did* account for using the medistation without a glow map and without the screen. Instead of a black shader or the armoury's screen, **one would need to use no shader or a transparent one for the screen to obtain what the artist had in mind!** I would still argue that this could have been designed better by using different colors (e.g. gray) for the diffusemap than for the glowmap.

**Comment by mjstahlberg** [comment]
What the artist had in mind for unpowered:

unvanquished_2023-04-08_090226_000

**Comment by mjstahlberg** [comment]
@illwieckz may I push to this branch first with a commit that produces the artist's intended behavior (screenshot above) and then possibly with an update to the diffuse map that improves upon it?

**Comment by mjstahlberg** [comment]
Work in progress/early draft on this:

Unpowered:
unvanquished_2023-04-08_095013_000

Powered:
unvanquished_2023-04-08_095247_000

**Comment by illwieckz** [comment]
> @illwieckz may I push to this branch first with a commit that produces the artist's intended behavior (screenshot above) and then possibly with an update to the diffuse map that improves upon it?

Yes please! That screenshot looks great!

**Comment by mjstahlberg** [comment]
Current state (more contrast/color variation than the draft above and trying to make the 3D projector appear as one):

unvanquished_2023-04-08_152829_000
unvanquished_2023-04-08_153236_000

**Comment by illwieckz** [comment]
Awesome! I love it!

**Comment by mjstahlberg** [comment]
Please cross-check if my commits are good and merge if you feel like it. :)

**Comment by ghost** [comment]
The glow in the middle of powered medi seems a bit weak though, it have a weird dark shadow around it, as if the medi was nearing lack of power.

Still: Much better than current situation or proposals, yes.

**Comment by mjstahlberg** [comment]
> The glow in the middle of powered medi seems a bit weak though, it have a weird dark shadow around it, as if the medi was nearing lack of power.

Can you provide a screenshot? The idea behind the changed material is that the spot in the middle is a projector which throws the 3D cross, so I tried to make it a bit like a contemporary beamer.

**Comment by illwieckz** [comment]
@Viech is `models/buildables/medistat/display-u.png` just a pure black texture?

**Comment by illwieckz** [comment]
> @Viech is `models/buildables/medistat/display-u.png` just a pure black texture?

It indeed looks like this. I optimized the material by using an engine builtin black texture, and by using the basic colormap GLSL shader instead of the complex lightMapping one (no need to compute the lightgrid for a fully transparent texture):

```
models/buildables/medistat/displayU
{
	{
		map $blackImage
		blendfunc add
	}
}
```

**Comment by mjstahlberg** [comment]
> @Viech is `models/buildables/medistat/display-u.png` just a pure black texture?

Yes. I wasn't sure if there was a black/transparent shader that one could safely use.

**Comment by illwieckz** [comment]
> > @Viech is `models/buildables/medistat/display-u.png` just a pure black texture?
> 
> Yes. I wasn't sure if there was a black/transparent shader that one could safely use.

If the material is meant to be seen in level editor and/or used by the map compiler, one can use `textures/shared_colors_src/black_d`, if the material is only meant to be seen in engine, one can use `$blackImage`.

**Comment by mjstahlberg** [comment]
From my end this resolves Unvanquished/Unvanquished#2598, anything I should improve or add; is the holographic-projector interpretation of the center spot acceptable?

**Comment by illwieckz** [comment]
> is the holographic-projector interpretation of the center spot acceptable?

Far better than before, before I thought it was just a simple light under a colored plastic, which was totally not compatible with a projector.

**Pull request merged**